Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelPage.xaml.cs

```diff
@@ -52,17 +53,75 @@ public APIModelPage(ApplicationAPIModel applicationAPIModelBase)
             SecondRow.MaxHeight = System.Windows.SystemParameters.PrimaryScreenHeight - 380;
 
             WorkSpace.Instance.RefreshGlobalAppModelParams(mApplicationAPIModel);
-            page = new ModelParamsPage(mApplicationAPIModel);
+            page = new ModelParamsPage(mApplicationAPIModel, viewMode);
             xDynamicParamsFrame.Content = page;
 
-            OutputTemplatePage outputTemplatePage = new OutputTemplatePage(mApplicationAPIModel);
+            OutputTemplatePage outputTemplatePage = new OutputTemplatePage(mApplicationAPIModel, viewMode);
             xOutputTemplateFrame.Content = outputTemplatePage;
 
             mApplicationAPIModel.AppModelParameters.CollectionChanged += AppModelParameters_CollectionChanged;
             mApplicationAPIModel.GlobalAppModelParameters.CollectionChanged += AppModelParameters_CollectionChanged;
             UpdateModelParametersTabHeader();
             mApplicationAPIModel.ReturnValues.CollectionChanged += ReturnValues_CollectionChanged;
             UpdateOutputTemplateTabHeader();
+
+            mPageViewMode = viewMode;
+
+            if (mPageViewMode == General.eRIPageViewMode.View)
+                UpdatePageAsReadOnly();
+
+            if (mPageViewMode == General.eRIPageViewMode.Add)
```

Comment:
what is "Add" view for API model page? why HttpHeadersGrid behave diffrently in it?

Reply:
Add for the Merger section over the CompareMergePage as to show and allow the Paste operation there.